compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/test/roundtrip.mlir

```diff
@@ -53,3 +65,40 @@ func.func @dma_cpy_nd_mixed(%arg0: !amdaie.logicalobjectfifo<memref<1x1x8x16xi32
   %0 = amdaie.dma_cpy_nd(%arg0[%c0, %c0, %c0, %c0] [1, 1, %c8, %c16] [%c128, %c128, %c16, 1], %arg1[%c0, %c0, %c0, %c0] [1, 1, %c8, %c16] [%c128, %c16, %c16, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 1>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
   return
 }
+
+// -----
+
+// CHECK-LABEL: func.func @logicalobjectfifo_consume
+// CHECK: amdaie.dma_cpy_nd
+// CHECK: amdaie.logicalobjectfifo.consume
+func.func @logicalobjectfifo_consume(%arg0: !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 1>>, %arg1: !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>) {
+  %0 = amdaie.dma_cpy_nd(%arg0[0, 0, 0, 0] [1, 1, 8, 16] [128, 128, 16, 1], %arg1[0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 1>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+  amdaie.logicalobjectfifo.consume(%0)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: func.func @logicalobjectfifo_produce
+// CHECK: amdaie.dma_cpy_nd
+// CHECK: amdaie.logicalobjectfifo.produce
+func.func @logicalobjectfifo_produce(%arg0: !amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 1>>, %arg1: !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>) {
+  %0 = amdaie.dma_cpy_nd(%arg0[0, 0, 0, 0] [1, 1, 8, 16] [128, 128, 16, 1], %arg1[0, 0, 0, 0] [1, 1, 8, 16] [128, 16, 16, 1]) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32, 1>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+  amdaie.logicalobjectfifo.produce(%0)
+  return
+}
+
+// -----
+
+// CHECK-LABEL: func.func @workgroup
+// CHECK: amdaie.workgroup
+// CHECK: amdaie.controlcode
+// CHECK: amdaie.end
+func.func @workgroup() {
+  amdaie.workgroup {
+    amdaie.controlcode {
+      amdaie.end
+    }
+  }
+  return
+}
```

Comment:
Why not include `amdaie.core`?

Reply:
I had a separate test for that, but added it here as well.